# Réalisez une étude santé publique

In [1]:
## Question 1

In [2]:
import pandas as pd

In [3]:
import numpy as np

In [4]:
df1 = []

In [5]:
df1 = pd.read_csv("Q1.csv", usecols = ["Pays", "Valeur"])

In [6]:
df1

Pays   Valeur
0                               Afghanistan    30552
1                            Afrique du Sud    52776
2                                   Albanie     3173
3                                   Algérie    39208
4                                 Allemagne    82727
5                                    Angola    21472
6                        Antigua-et-Barbuda       90
7                           Arabie saoudite    28829
8                                 Argentine    41446
9                                   Arménie     2977
10                                Australie    23343
11                                 Autriche     8495
12                              Azerbaïdjan     9413
13                                  Bahamas      377
14                               Bangladesh   156595
15                                  Barbade      285
16                                  Bélarus     9357
17                                 Belgique    11104
18                                   Belize      332
19                                    Bénin    10323
20                                 Bermudes       65
21          Bolivie (État plurinational de)    10671
22                       Bosnie-Herzégovine     3829
23                                 Botswana     2021
24                                   Brésil   200362
25                        Brunéi Darussalam      418
26                                 Bulgarie     7223
27                             Burkina Faso    16935
28                               Cabo Verde      499
29                                 Cambodge    15135
..                                      ...      ...
145                                 Sénégal    14133
146                                  Serbie     9511
147                            Sierra Leone     6092
148                               Slovaquie     5450
149                                Slovénie     2072
150                                  Soudan    37964
151                               Sri Lanka    21273
152                                   Suède     9571
153                                  Suisse     8078
154                                Suriname      539
155                                Eswatini     1250
156                             Tadjikistan     8208
157                                   Tchad    12825
158                           Tchéquie (la)    10702
159                               Thaïlande    67011
160                             Timor-Leste     1133
161                                    Togo     6817
162                       Trinité-et-Tobago     1341
163                                 Tunisie    10997
164                            Turkménistan     5240
165                                 Turquie    74933
166                                 Ukraine    45239
167                                 Uruguay     3407
168                                 Vanuatu      253
169  Venezuela (République bolivarienne du)    30405
170                                Viet Nam    91680
171                                   Yémen    24407
172                                  Zambie    14539
173                                Zimbabwe    14150
174                                   Chine  1416667

[175 rows x 2 columns]

In [7]:
df1['Valeur'].sum() * 1000

8413993000

- Ce chiffre est supérieur à la population mondiale officielle. Il englobe deux fois la chine continentale.

In [8]:
df1_wo_china = df1[df1['Pays'] != "Chine"]

In [9]:
df1_wo_china['Valeur'].sum() * 1000

6997326000

- C'est une valeur plus conforme aux chiffres officiels (Source: https://www.prb.org/wp-content/uploads/2014/02/2013-population-data-sheet_french.pdf)

In [10]:
temp = pd.read_csv("pop.csv")

In [11]:
temp.columns = ['xx','xx2','country_code','country','xx3','element'
    ,'item_code','item','xx4',"year","unit","value",'xx5','xx6']

In [12]:
pop = temp.pivot_table(
    index=["country_code","country","item_code","item","year"],
    columns = ["element"], values=["value"], aggfunc=sum)

In [13]:
pop.columns = ['total_population']

In [14]:
pop.reset_index()

country_code                             country  item_code        item  \
0               1                             Armenia       2501  Population   
1               2                         Afghanistan       2501  Population   
2               3                             Albania       2501  Population   
3               4                             Algeria       2501  Population   
4               7                              Angola       2501  Population   
5               8                 Antigua and Barbuda       2501  Population   
6               9                           Argentina       2501  Population   
7              10                           Australia       2501  Population   
8              11                             Austria       2501  Population   
9              12                             Bahamas       2501  Population   
10             14                            Barbados       2501  Population   
11             16                          Bangladesh       2501  Population   
12             17                             Bermuda       2501  Population   
13             19    Bolivia (Plurinational State of)       2501  Population   
14             20                            Botswana       2501  Population   
15             21                              Brazil       2501  Population   
16             23                              Belize       2501  Population   
17             25                     Solomon Islands       2501  Population   
18             26                   Brunei Darussalam       2501  Population   
19             27                            Bulgaria       2501  Population   
20             28                             Myanmar       2501  Population   
21             32                            Cameroon       2501  Population   
22             33                              Canada       2501  Population   
23             35                          Cabo Verde       2501  Population   
24             37            Central African Republic       2501  Population   
25             38                           Sri Lanka       2501  Population   
26             39                                Chad       2501  Population   
27             40                               Chile       2501  Population   
28             41                     China, mainland       2501  Population   
29             44                            Colombia       2501  Population   
..            ...                                 ...        ...         ...   
144           210                              Sweden       2501  Population   
145           211                         Switzerland       2501  Population   
146           213                        Turkmenistan       2501  Population   
147           214           China, Taiwan Province of       2501  Population   
148           215         United Republic of Tanzania       2501  Population   
149           216                            Thailand       2501  Population   
150           217                                Togo       2501  Population   
151           220                 Trinidad and Tobago       2501  Population   
152           221                                Oman       2501  Population   
153           222                             Tunisia       2501  Population   
154           223                              Turkey       2501  Population   
155           225                United Arab Emirates       2501  Population   
156           226                              Uganda       2501  Population   
157           229                      United Kingdom       2501  Population   
158           230                             Ukraine       2501  Population   
159           231            United States of America       2501  Population   
160           233                        Burkina Faso       2501  Population   
161           234                             Uruguay       2501  Population   
162  

## Question 2

In [15]:
df2 = pd.read_csv("Q2.csv")

In [16]:
df2

Code Domaine              Domaine  Code Pays    Pays  Code Élément  \
0           FBS  Bilans Alimentaires         68  France          5511   
1           FBS  Bilans Alimentaires         68  France          5611   
2           FBS  Bilans Alimentaires         68  France          5072   
3           FBS  Bilans Alimentaires         68  France          5911   
4           FBS  Bilans Alimentaires         68  France          5301   
5           FBS  Bilans Alimentaires         68  France          5521   
6           FBS  Bilans Alimentaires         68  France          5527   
7           FBS  Bilans Alimentaires         68  France          5123   
8           FBS  Bilans Alimentaires         68  France          5131   
9           FBS  Bilans Alimentaires         68  France          5154   
10          FBS  Bilans Alimentaires         68  France          5142   
11          FBS  Bilans Alimentaires         68  France           645   
12          FBS  Bilans Alimentaires         68  France           664   
13          FBS  Bilans Alimentaires         68  France           674   
14          FBS  Bilans Alimentaires         68  France           684   

                                              Élément  Code Produit Produit  \
0                                          Production          2511     Blé   
1                             Importations - Quantité          2511     Blé   
2                                  Variation de stock          2511     Blé   
3                             Exportations - Quantité          2511     Blé   
4                            Disponibilité intérieure          2511     Blé   
5                               Aliments pour animaux          2511     Blé   
6                                            Semences          2511     Blé   
7                                              Pertes          2511     Blé   
8                                          Traitement          2511     Blé   
9                                 Autres Utilisations          2511     Blé   
10                                         Nourriture          2511     Blé   
11  Disponibilité alimentaire en quantité (kg/pers...          2511     Blé   
12     Disponibilité alimentaire (Kcal/personne/jour)          2511     Blé   
13  Disponibilité de protéines en quantité (g/pers...          2511     Blé   
14  Disponibilité de matière grasse en quantité (g...          2511     Blé   

    Code Année  Année               Unité    Valeur Symbole  \
0         2013   2013  Milliers de tonnes  38614.00       S   
1         2013   2013  Milliers de tonnes   2055.00       S   
2         2013   2013  Milliers de tonnes   1131.00       S   
3         2013   2013  Milliers de tonnes  21502.00       S   
4         2013   2013  Milliers de tonnes  20298.00       S   
5         2013   2013  Milliers de tonnes   7822.00       S   
6         2013   2013  Milliers de tonnes    748.00       S   
7         2013   2013  Milliers de tonnes    358.00       S   
8         2013   2013  Milliers de tonnes   1575.00       S   
9         2013   2013  Milliers de tonnes   2824.00       S   
10        2013   2013  Milliers de tonnes   6971.00       S   
11        2013   2013                  kg    108.43      Fc   
12        2013   2013  Kcal/personne/jour    811.00      Fc   
13        2013   2013     g/personne/jour     25.99      Fc   
14        2013   2013     g/personne/jour      4.20      Fc   

   Description du Symbole  
0   Données standardisées  
1   Données standardisées  
2   Données standardisées  
3   Données standardisées  
4   Données standardisées  
5   Données standardisées  
6   Données standardisées  
7   Données standardisées  
8   Données standardisées  
9   Données standardisées  
10  Données standardisées  
11        Donnée calculée  
12        Donnée calculée  
13        Donnée calculée  
14        Donnée calculée

- Disponibilité intérieure = Production + Importations - Exportations + Variation des stocks = Nourriture + Aliments pour animaux + Semences + Pertes + Traitement + Autres Utilisations

## Question 3

In [17]:
# Import des données des bilans alimentaires
veg = pd.read_csv("veg.csv")
ani = pd.read_csv("ani.csv")

# Ajout de la variable origin
ani["origin"] = "animal"
veg["origin"] = "vegetal"

# On regroupe veg et ani en un unique dataframe, via une union
temp = ani.append(veg)

# Suppression de ani et veg
del ani, veg

# On renomme les colonnes de temp
temp.columns = ["xx","xx2","country_code","country",'xx3','element'
    ,'item_code','item','xx4',"year","unit","value",'xx5','xx6'
    ,'origin']

# Transformation de temp en table pivot
data = temp.pivot_table(
    index=["country_code","country","item_code","item","year","origin"],
    columns = ["element"], values=["value"], aggfunc=sum)
# On renomme les colonnes (attention l'ordre peut changer selon vos données !)
data.columns = ['domestic_supply_quantity', 'export_quantity', 'fat_supply_quantity_gcapitaday',
    'feed', 'food', 'food_supply_kcalcapitaday', 
    'food_supply_quantity_kgcapitayr', 'import_quantity', 'waste', 'other_uses',
    'processing', 'production', 'protein_supply_quantity_gcapitaday', 
    'seed', 'stock_variation']
    
data = data.reset_index()
data.head()

country_code  country  item_code                 item  year   origin  \
0             1  Armenia       2511   Wheat and products  2013  vegetal   
1             1  Armenia       2513  Barley and products  2013  vegetal   
2             1  Armenia       2514   Maize and products  2013  vegetal   
3             1  Armenia       2515     Rye and products  2013  vegetal   
4             1  Armenia       2516                 Oats  2013  vegetal   

   domestic_supply_quantity  export_quantity  fat_supply_quantity_gcapitaday  \
0                     554.0              1.0                            3.60   
1                     198.0              0.0                            0.00   
2                     102.0              NaN                             NaN   
3                       1.0              NaN                            0.00   
4                       6.0              NaN                            0.03   

    feed  ...  food_supply_kcalcapitaday  food_supply_quantity_kgcapitayr  \
0   93.0  ...                     1024.0                           130.60   
1  137.0  ...                        0.0                             0.00   
2   96.0  ...                        0.0                             0.03   
3    1.0  ...                        1.0                             0.12   
4    4.0  ...                        2.0                             0.37   

   import_quantity  waste  other_uses  processing  production  \
0            361.0   32.0         0.0        10.0       312.0   
1              9.0   15.0        26.0         7.0       189.0   
2             82.0    7.0         NaN         NaN        21.0   
3              0.0    0.0         NaN         NaN         1.0   
4              1.0    0.0         NaN         NaN         5.0   

   protein_supply_quantity_gcapitaday  seed  stock_variation  
0                               30.52  30.0           -118.0  
1                                0.00  14.0              0.0  
2                                0.01   0.0              NaN  
3                                0.02   0.0              0.0  
4                                0.09   0.0              NaN  

[5 rows x 21 columns]

In [18]:
data = data.merge(pop, on=['country'])

In [19]:
data['total_kcal'] = data['food_supply_kcalcapitaday'] * data['total_population'] * 1000 * 365
data.head()

country_code  country  item_code                 item  year   origin  \
0             1  Armenia       2511   Wheat and products  2013  vegetal   
1             1  Armenia       2513  Barley and products  2013  vegetal   
2             1  Armenia       2514   Maize and products  2013  vegetal   
3             1  Armenia       2515     Rye and products  2013  vegetal   
4             1  Armenia       2516                 Oats  2013  vegetal   

   domestic_supply_quantity  export_quantity  fat_supply_quantity_gcapitaday  \
0                     554.0              1.0                            3.60   
1                     198.0              0.0                            0.00   
2                     102.0              NaN                             NaN   
3                       1.0              NaN                            0.00   
4                       6.0              NaN                            0.03   

    feed  ...  import_quantity  waste  other_uses  processing  production  \
0   93.0  ...            361.0   32.0         0.0        10.0       312.0   
1  137.0  ...              9.0   15.0        26.0         7.0       189.0   
2   96.0  ...             82.0    7.0         NaN         NaN        21.0   
3    1.0  ...              0.0    0.0         NaN         NaN         1.0   
4    4.0  ...              1.0    0.0         NaN         NaN         5.0   

   protein_supply_quantity_gcapitaday  seed  stock_variation  \
0                               30.52  30.0           -118.0   
1                                0.00  14.0              0.0   
2                                0.01   0.0              NaN   
3                                0.02   0.0              0.0   
4                                0.09   0.0              NaN   

   total_population    total_kcal  
0              2977  1.112684e+12  
1              2977  0.000000e+00  
2              2977  0.000000e+00  
3              2977  1.086605e+09  
4              2977  2.173210e+09  

[5 rows x 23 columns]

In [20]:
data['total_protein_kg'] = data['protein_supply_quantity_gcapitaday'] * 1000 * 365 * data['total_population'] * 1000
data.head()

country_code  country  item_code                 item  year   origin  \
0             1  Armenia       2511   Wheat and products  2013  vegetal   
1             1  Armenia       2513  Barley and products  2013  vegetal   
2             1  Armenia       2514   Maize and products  2013  vegetal   
3             1  Armenia       2515     Rye and products  2013  vegetal   
4             1  Armenia       2516                 Oats  2013  vegetal   

   domestic_supply_quantity  export_quantity  fat_supply_quantity_gcapitaday  \
0                     554.0              1.0                            3.60   
1                     198.0              0.0                            0.00   
2                     102.0              NaN                             NaN   
3                       1.0              NaN                            0.00   
4                       6.0              NaN                            0.03   

    feed  ...  waste  other_uses  processing  production  \
0   93.0  ...   32.0         0.0        10.0       312.0   
1  137.0  ...   15.0        26.0         7.0       189.0   
2   96.0  ...    7.0         NaN         NaN        21.0   
3    1.0  ...    0.0         NaN         NaN         1.0   
4    4.0  ...    0.0         NaN         NaN         5.0   

   protein_supply_quantity_gcapitaday  seed  stock_variation  \
0                               30.52  30.0           -118.0   
1                                0.00  14.0              0.0   
2                                0.01   0.0              NaN   
3                                0.02   0.0              0.0   
4                                0.09   0.0              NaN   

   total_population    total_kcal  total_protein_kg  
0              2977  1.112684e+12      3.316318e+13  
1              2977  0.000000e+00      0.000000e+00  
2              2977  0.000000e+00      1.086605e+10  
3              2977  1.086605e+09      2.173210e+10  
4              2977  2.173210e+09      9.779445e+10  

[5 rows x 24 columns]

## Question 4

In [21]:
data['kcal/kg'] = data['food_supply_kcalcapitaday'] * 365 / data['food_supply_quantity_kgcapitayr']
data.head()

country_code  country  item_code                 item  year   origin  \
0             1  Armenia       2511   Wheat and products  2013  vegetal   
1             1  Armenia       2513  Barley and products  2013  vegetal   
2             1  Armenia       2514   Maize and products  2013  vegetal   
3             1  Armenia       2515     Rye and products  2013  vegetal   
4             1  Armenia       2516                 Oats  2013  vegetal   

   domestic_supply_quantity  export_quantity  fat_supply_quantity_gcapitaday  \
0                     554.0              1.0                            3.60   
1                     198.0              0.0                            0.00   
2                     102.0              NaN                             NaN   
3                       1.0              NaN                            0.00   
4                       6.0              NaN                            0.03   

    feed  ...  other_uses  processing  production  \
0   93.0  ...         0.0        10.0       312.0   
1  137.0  ...        26.0         7.0       189.0   
2   96.0  ...         NaN         NaN        21.0   
3    1.0  ...         NaN         NaN         1.0   
4    4.0  ...         NaN         NaN         5.0   

   protein_supply_quantity_gcapitaday  seed  stock_variation  \
0                               30.52  30.0           -118.0   
1                                0.00  14.0              0.0   
2                                0.01   0.0              NaN   
3                                0.02   0.0              0.0   
4                                0.09   0.0              NaN   

   total_population    total_kcal  total_protein_kg      kcal/kg  
0              2977  1.112684e+12      3.316318e+13  2861.868300  
1              2977  0.000000e+00      0.000000e+00          NaN  
2              2977  0.000000e+00      1.086605e+10     0.000000  
3              2977  1.086605e+09      2.173210e+10  3041.666667  
4              2977  2.173210e+09      9.779445e+10  1972.972973  

[5 rows x 25 columns]

In [22]:
temp = data[data['item'].str.contains("Eggs")]

In [23]:
# valeur nutritionnelle pour 100g
temp['kcal/kg'].mean()/10

136.48177721733776

- ce chiffre est proche de la valeur de 147 kcal pour un oeuf de poule entier cru donné par Wikipedia (https://fr.wikipedia.org/wiki/Œuf_(aliment))

In [24]:
data['protein_percentage'] = data['protein_supply_quantity_gcapitaday'] / data['food_supply_quantity_kgcapitayr'] * 100

In [25]:
data.head()

country_code  country  item_code                 item  year   origin  \
0             1  Armenia       2511   Wheat and products  2013  vegetal   
1             1  Armenia       2513  Barley and products  2013  vegetal   
2             1  Armenia       2514   Maize and products  2013  vegetal   
3             1  Armenia       2515     Rye and products  2013  vegetal   
4             1  Armenia       2516                 Oats  2013  vegetal   

   domestic_supply_quantity  export_quantity  fat_supply_quantity_gcapitaday  \
0                     554.0              1.0                            3.60   
1                     198.0              0.0                            0.00   
2                     102.0              NaN                             NaN   
3                       1.0              NaN                            0.00   
4                       6.0              NaN                            0.03   

    feed  ...  processing  production  protein_supply_quantity_gcapitaday  \
0   93.0  ...        10.0       312.0                               30.52   
1  137.0  ...         7.0       189.0                                0.00   
2   96.0  ...         NaN        21.0                                0.01   
3    1.0  ...         NaN         1.0                                0.02   
4    4.0  ...         NaN         5.0                                0.09   

   seed  stock_variation  total_population    total_kcal  total_protein_kg  \
0  30.0           -118.0              2977  1.112684e+12      3.316318e+13   
1  14.0              0.0              2977  0.000000e+00      0.000000e+00   
2   0.0              NaN              2977  0.000000e+00      1.086605e+10   
3   0.0              0.0              2977  1.086605e+09      2.173210e+10   
4   0.0              NaN              2977  2.173210e+09      9.779445e+10   

       kcal/kg  protein_percentage  
0  2861.868300           23.369066  
1          NaN                 NaN  
2     0.000000           33.333333  
3  3041.666667           16.666667  
4  1972.972973           24.324324  

[5 rows x 26 columns]

In [26]:
temp = data[data['item'].str.contains("Oats")]

In [27]:
temp = temp[temp['protein_supply_quantity_gcapitaday'] > 0]

In [28]:
temp['protein_percentage'].mean()

24.116556631276456

## Question 5

In [29]:
temp = data[data['kcal/kg'] > 0]
temp.pivot_table(
    index=["item_code","item","year","origin"],
    values=["kcal/kg"], aggfunc=np.mean).sort_values(by='kcal/kg', ascending=False).head(20)

kcal/kg
item_code item                 year origin               
2562      Palm kernels         2013 vegetal           inf
2558      Rape and Mustardseed 2013 vegetal           inf
2782      Fish, Liver Oil      2013 animal   1.180900e+04
2579      Sesameseed Oil       2013 vegetal  1.014606e+04
2580      Olive Oil            2013 vegetal  9.292674e+03
2582      Maize Germ Oil       2013 vegetal  9.257130e+03
2581      Ricebran Oil         2013 vegetal  9.206134e+03
2574      Rape and Mustard Oil 2013 vegetal  9.149964e+03
2572      Groundnut Oil        2013 vegetal  9.118263e+03
2573      Sunflowerseed Oil    2013 vegetal  9.027792e+03
2576      Palmkernel Oil       2013 vegetal  9.012118e+03
2575      Cottonseed Oil       2013 vegetal  8.972360e+03
2578      Coconut Oil          2013 vegetal  8.932572e+03
2781      Fish, Body Oil       2013 animal   8.917431e+03
2571      Soyabean Oil         2013 vegetal  8.900181e+03
2577      Palm Oil             2013 vegetal  8.791218e+03
2586      Oilcrops Oil, Other  2013 vegetal  8.720818e+03
2737      Fats, Animals, Raw   2013 animal   7.579279e+03
2740      Butter, Ghee         2013 animal   7.470526e+03
2561      Sesame seed          2013 vegetal  6.136984e+03

In [ ]:
- Parmi les 20 aliments les plus caloriques, on peut citer l'huile d'olive, le poisson, les graisses animales, le beurre et les graines de sesame

In [35]:
temp.pivot_table(
    index=["item_code","item","year","origin"],
    values=["protein_percentage"], aggfunc=np.mean).sort_values(by='protein_percentage', ascending=False).head(20)

protein_percentage
item_code item                       year origin                     
2558      Rape and Mustardseed       2013 vegetal          100.000000
2555      Soyabeans                  2013 vegetal           74.669990
2556      Groundnuts (Shelled Eq)    2013 vegetal           68.743316
2547      Peas                       2013 vegetal           61.435459
2549      Pulses, Other and products 2013 vegetal           61.108724
2546      Beans                      2013 vegetal           59.906316
2570      Oilcrops, Other            2013 vegetal           51.452553
2561      Sesame seed                2013 vegetal           51.262232
2735      Meat, Other                2013 animal            48.899040
2736      Offals, Edible             2013 animal            48.303837
2680      Infant food                2013 vegetal           42.278748
2766      Cephalopods                2013 animal            41.192422
2731      Bovine Meat                2013 animal            39.619474
2732      Mutton & Goat Meat         2013 animal            38.073761
2734      Poultry Meat               2013 animal            34.575385
2641      Pimento                    2013 vegetal           34.057608
2763      Pelagic Fish               2013 animal            33.623757
2557      Sunflower seed             2013 vegetal           33.040903
2764      Marine Fish, Other         2013 animal            31.110611
2640      Pepper                     2013 vegetal           30.852471

In [ ]:
- Parmi les 20 aliments les plus riches en protéine, on peut citer:
    les graines de soja, les noix, les poix, les mollusques, les viandes

## Question 6

In [ ]:
temp = data[data['origin'] == "vegetal"]

## Question 7

In [40]:
temp['total_kcal'].sum()/1000000000/2000/365

8.557396728999999

In [ ]:
- La disponibilité intérieure mondiale de produits végétaux permettrait de nourrir 8,5 milliards de personnes